base/twiceprecision.jl

```diff
@@ -535,7 +535,22 @@ function __convertSRL(::Type{StepRangeLen{T,R,S}}, r::AbstractRange{U}) where {T
     StepRangeLen{T,R,S}(R(first(r)), S(step(r)), length(r))
 end
 
-function sum(r::StepRangeLen)
+function sum(r::StepRangeLen{T,Float64,Float64}) where {T}
+    l = length(r)
+    # Compute the contribution of step over all indices.
+    # Indexes on opposite side of r.offset contribute with opposite sign,
+    #    r.step * (sum(1:np) - sum(1:nn))
+    np, nn = l - r.offset, r.offset - 1  # positive, negative
+    # To prevent overflow in sum(1:n), multiply its factors by the step
+    sp, sn = sumpair(np), sumpair(nn)
+    tp = r.step * sp[1] * sp[2]
```

Comment:
This will indeed be safe for `r.step isa AbstractFloat` but perhaps not if it's an integer. What about something like this?

```julia
W = widen(Int)
Δn = Int(W(sp[1]) * W(sp[2]) - W(sn[1]) * W(sn[2]))
s = r.ref * Δn
```

The length of `StepRangeLen` is hard-coded as `Int`, so this seems fairly safe.